### PR TITLE
CoverArtDelegate: Fix repeated repaint if covers could not be loaded 

### PR DESCRIPTION
**Pull request by uklotzde**
https://mixxx.zulipchat.com/#narrow/stream/109171-development/topic/high.20CPU.20use.20with.20cover.20art.20column

### CODE REVIEW

File: src/library/coverartdelegate.cpp [hunk 1]
```diff
@@ -43,15 +43,14 @@ CoverArtDelegate::CoverArtDelegate(QTableView* parent)
...
-    if (rows.isEmpty()) {
-        return;
-    }
     // Sort in ascending order...
     std::sort(rows.begin(), rows.end());
     // ...and then deduplicate...
     rows.erase(std::unique(rows.begin(), rows.end()), rows.end());
     // ...before emitting the signal.
-    DEBUG_ASSERT(!rows.isEmpty());
+    if (rows.isEmpty()) {
```
Comment (daschuer) [inline src/library/coverartdelegate.cpp:51]: Why is this changed? The commands in between are looking senseless in case of rows being empty. 
If they are there for a reason, please comment the source accordingly.

File: src/library/coverartdelegate.cpp [hunk 2]
```diff
@@ -63,11 +62,11 @@ void CoverArtDelegate::slotInhibitLazyLoading(
     }
     // If we can request non-cache covers now, request updates
     // for all rows that were cache misses since the last time.
-    auto staleRows = m_cacheMissRows;
     // Reset the member variable before mutating the aggregated
     // rows list (-> implicit sharing) and emitting a signal that
     // in turn may trigger new signals for CoverArtDelegate!
-    m_cacheMissRows = QList<int>();
+    QList<int> staleRows = std::move(m_cacheMissRows);
```
Comment (daschuer) [inline src/library/coverartdelegate.cpp:68]: I think we can get rid of intermediate staleRows. I am unsure if the compiler is smart enough to optimize it away.

**Review by daschuer — CHANGES_REQUESTED** [review]
Thank you. I am glad you have found the issue. 

Can this also be this issue for this one? 
https://bugs.launchpad.net/mixxx/+bug/1905124

File: src/library/coverartdelegate.cpp (hunk 1 above)
Comment (uklotzde) [reply src/library/coverartdelegate.cpp:51]: You are right. Can't become empty when deduplicated. I'll revert and force push.

File: src/library/coverartdelegate.cpp (hunk 2 above)
Comment (uklotzde) [reply src/library/coverartdelegate.cpp:68]: Sure. But I wanted to keep the debug assertion, just in case.

**Comment by uklotzde** [comment]
> Can this also be this issue for this one?
> https://bugs.launchpad.net/mixxx/+bug/1905124

I don't think so. Mixxx was still usable even with the bug.

File: src/library/coverartdelegate.cpp (hunk 2 above)
Comment (daschuer) [reply src/library/coverartdelegate.cpp:69]: It can be moved below emitRowsChanged()

**Review by daschuer — APPROVED** [review]
LGTM
@ronso0 are you able to confirm that this fixes you issue?

**Comment by ronso0** [comment]
Yes, excellent, can't provoke the CPU overload anymore.
Thank you for the quick fix @uklotzde

**Review by ronso0 — APPROVED** [review]
